dbms/src/Interpreters/Context.cpp

```diff
@@ -1423,13 +1423,24 @@ QueryLog & Context::getQueryLog()
 
         auto & config = getConfigRef();
 
-        String database = config.getString("query_log.database", "system");
-        String table = config.getString("query_log.table", "query_log");
+        String database     = config.getString("query_log.database",     "system");
+        String table        = config.getString("query_log.table",        "query_log");
+        String partition_by = config.getString("query_log.partition_by", "month");
         size_t flush_interval_milliseconds = config.getUInt64(
                 "query_log.flush_interval_milliseconds", DEFAULT_QUERY_LOG_FLUSH_INTERVAL_MILLISECONDS);
+        
+        String engine;
+
+        if (partition_by == "day")
```

Comment:
Maybe it's better to read contents of PARTITION BY directly from config?
example: `<partition_by>toMonday(event_date)</partition_by>`

We can always add event_date to primary key as it doesn't make anything worse for day partitioning.

Reply:
OK, I'll change code and docs